src/util.js

```diff
@@ -0,0 +1,82 @@
+import Ajv from "ajv";
+
+const STAC_PROXY_URL =
+  process.env.STAC_PROXY_URL;
+
+export async function fetchUri(uri) {
+  // If we are proxying a STAC Catalog, replace any URI with the proxied address.
+  // STAC_PROXY_URL has the form https://thingtoproxy.com|http://proxy:111
+  const proxiedUri = !!STAC_PROXY_URL ? (
+    uri.replace(STAC_PROXY_URL.split('|')[0], STAC_PROXY_URL.split('|')[1])
+  ) : uri;
+  return fetch(proxiedUri);
+};
+
+const modifyLoadSchemaUri = function(baseUrl, uri) {
+  if(uri.includes("://")) { return uri; } // Absolute URI
+
+  if(uri.includes("/")) {
+    // Relative path, e.g. collection to catalog.
+    return `${baseUrl}/${uri}`;
+  }
+
+  // Common Metadata
+  return `${baseUrl}/item-spec/json-schema/${uri}`;
+};
+
+const fixUpSchema = function(schema, schemaUri) {
+  // Make $id unique, otherwise AjV will complain
+  schema.$id = schemaUri + '#';
+
+  // Fix old schemas that have 'id' instead of '$id'
+  if("id" in schema) {
```

Comment:
Do you really need to do that yourself or could you just set the ajv option `schemaId` to `auto`?

Reply:
Hm, good call, but I tried this but was still getting an error

```
Uncaught (in promise) Error: schema $id is different from id
    at Ajv._get$IdOrId [as _getId] (ajv.js:398)
    at Ajv._addSchema (ajv.js:300)
    at async.js:31
```

Not sure why this is happening base on the Ajv docs; perhaps it's based on the secondary schemas loaded in that are somehow not affected by the options? Either way, I'll leave in this workaround for now.